File: trunk/JAVA/jzbus/src/test/java/net/zbus/asyn/ClientRecv.java
package net.zbus.asyn;

import net.zbus.BusClient;
import net.zbus.ConnectionConfig;
import net.zbus.ZMsg;

public class ClientRecv {

	public static void main(String[] args) {
		ConnectionConfig config = new ConnectionConfig();
		config.setId("local_mq");
		BusClient client = new BusClient(config);

		while (true) {
			try {
				ZMsg res = client.recv(2000);
				res.dump();
			} catch (Exception e) {
				e.printStackTrace();
				break;
			}
		}

		client.destroy();
	}

}

File: trunk/MS.NET/clrzbus/examples/asyn/ClientRecv.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class ClientRecv
    {
        public static void Main(string[] args)
        {
            ConnectionConfig connCfg = new ConnectionConfig();
            connCfg.Host = "127.0.0.1";
            connCfg.Port = 15555;
            connCfg.Id = "local_mq";

            BusClient client = new BusClient(connCfg);

            while (true)
            {
                try
                { 
                    ZMsg msg = client.Recv(4000);//4s for ping
                    msg.Dump();
                }
                catch (Exception e)
                {
                    Console.WriteLine(e);
                    break;
                }
            }

            client.Destroy();
           
        } 
    }
}
